api/objects.proto

```diff
@@ -342,6 +342,15 @@ message Cluster {
 	// reject joining the cluster.  Nodes that report themselves to be non-FIPS
 	// should be rejected from the cluster.
 	bool fips = 10 [(gogoproto.customname) = "FIPS"];
+
+	// This field specifies default subnet pools for global scope networks. If
+	// unspecified, Docker will use the predefined subnets as it works on older releases.
+	// Format Example : {"20.20.0.0/16",""20.20.0.0/16"}
+	repeated string defaultAddressPool = 11;
+
+	// This flag specifies the default subnet size of global scope networks by giving
+	// the length of the subnet masks for every such network
+	uint32 subnetSize = 12;
```

Comment:
@selansen I saw the earlier discussion about the naming of this field; you mentioned that you named it `subnetSize` because it was named like that in the existing local address pools (daemon option); do you know if that's the only place that name is exposed to users (i.e., if it's not visible in other locations, e.g. in the API)?

Because if that's the only place, we _could_ still decide to change; possibly deprecate the daemon option and/or accept both the new and old name for backward compatibility.

TL;DR if `subnetSize` is confusing, and we _have_ a chance to fix that confusing name, we should definitely consider doing so

Reply:
tbh; I was still on the fence for combining subnet and size for each pool, and (if needed) enforce hem to be all the same size on the CLI, but (technically) allow them to have different sizes)

Reply:
We use same name across CLI->MOBY . MOBY->SwarmKit and of course the old feature local scope default address pool.  

Regarding allowing user to have different subnet size for each subnet pool was modified after realizing the complications it brought to libnetwork.  Else we could have had same syntax for old address pool feature. 
But forcing user to provide same subnet pool for each address pool will be repetitive for user.

Reply:
> But forcing user to provide same subnet pool for each address pool will be repetitive for user.

That's UX, that's still separate from API.

> the complications it brought to libnetwork

Still curious what the complication was 😅 (because we send a separate size for each pool to libnetwork (but they happen to all be the same size)

Reply:
I thought you meant letting user to enter same size for each subnet pool. I didnt mean combining them in API.  LOL. my bad.  For this no complications in libnetwork